### DISCUSSION TITLE
replace unrenderable custom tag with new content or partial?

### DISCUSSION THREAD

**Discussion by paulrudy** [markdoc/markdoc#453] [Q&A] [answered]
I'm still fairly unfamilar with Markdoc, and hope this is a fairly simple question to answer.

I'm using Astro, and I have some custom tags defined in Astro's `markdoc.config.mjs` that render Astro components when the site is rendered.

But for rendering an RSS feed, Astro doesn't provide a way to access its render pipeline, so I can't render those components into the RSS post content.

What I'd like to do is replace the nodes containing those Astro components with a placeholder like `<p>Please view the <a href="https://some-site.org">online version</a> of this post to view this content</p>`

In my RSS endpoint in Astro, I have the following (`defineMarkdocConfig` is the Markdoc config as compiled by Astro):

```js
const renderMarkdoc = (body) => {
// replace Markdoc custom tags that call Astro components, until/unless there is a way to render them
const configWithoutComponents = { ...defineMarkdocConfig };
for (let tag in configWithoutComponents.tags) {
  if (configWithoutComponents.tags[tag]?.render?.type === 'package') {
    configWithoutComponents.tags[tag] = {
      ...configWithoutComponents.tags[tag],
      ...{
        transform() {
          return new Markdoc.Tag('p');
        },
      },
    };
  }
}
const ast = Markdoc.parse(body);
const content = Markdoc.transform(ast, configWithoutComponents);

const html = Markdoc.renderers.html(content);

return html;
};
```

Thanks in advance for any help and suggestions.
As written, this replaces the unrenderable Astro components with `<p></p>`. How can I include the full replacement text I quoted above? I thought about creating a partial, but it looks like there's no way to use `new Markdoc.Partial()` as a constructor.

**Comment by rpaul-stripe** [comment, marked as answer]
The `Markdoc.Tag` constructor works a bit like React's `createElement`: the first argument is the tag name, the second argument is an object with all of the tag's attributes, and the third argument is an array of children. Plain text nodes are represented with strings. Here's an example:

```javascript
new Markdoc.Tag('p', {}, [
  'Please view the ',
  new Markdoc.Tag('a', {href: 'http://some-site.org'}, ['online version']),
  ' of this post to view this content.'
])
```

**Reply by paulrudy** [reply]
Thank you so much @rpaul-stripe for the quick response! Just what I needed 🚀
